xcodeproj/internal/bazel_integration_files/process_bazel_build_log.py

```diff
@@ -75,6 +76,9 @@ def _process_log_line(line: str):
             return
 
         output_line = relative_diagnostic.sub(_replacement, input_line)
+        if not has_formatted_error and " error: " in output_line:
+            has_formatted_error = True
```

Comment:
Can't we use `relative_diagnostic` in some way to do the substitution and know if we did it? So if we did the substitution, we know that there was an error or warning. Maybe check `input_line != output_line` (if there isn't a better way)?

Reply:
I thought about it, but the regex also matches warnings, and we only want to show this error if there are no errors at all (regardless if there are warnings or not).

Reply:
Hmm. If there are warnings, does it show the weird exit code thing?

Reply:
I see what you mean, and yes you're right. We can simply check the exit code and see if we have previously done a substitution for a relative diagnostic to see if we need to print the default error message.